pandas/_typing.py

```diff
@@ -1,4 +1,11 @@
 from pathlib import Path
-from typing import IO, AnyStr, Union
+from typing import IO, AnyStr, Type, Union
 
+import numpy as np
+
+from pandas.core.dtypes.dtypes import ExtensionDtype
+from pandas.core.dtypes.generic import ABCExtensionArray
+
+ArrayLike = Union[ABCExtensionArray, np.ndarray]
+Dtype = Union[str, np.dtype, ExtensionDtype, Type]
```

Comment:
Hmm I wonder if `Type` should be more strict, i.e. just `Union[float, int, object]`

Reply:
Do you mean replacing Line 10 with `Dtype = Union[str, np.dtype, ExtensionDtype, float, int, object]`?

Reply:
Yea I think being stricter / more explicit with that would be preferable

Reply:
In that case, shouldn't we use `Union[str, np.dtype, ExtensionDtype, Type[float, int, object]]` since we're expecting `float`, `int`, and `object` types explicitly rather than instances of those types? E.g., we're not expecting SparseDtype to take a kw argument `dtype = 3`, but we would be ok with `dtype = int`.

It's worth noting that the SparseDtype doc explicitly states that dtype will take `type`, so restricting our new `Dtype` (so far only used in that class) tighter takes it out of line with the docs.

That being said, I think it's in keeping with the goal of type checking to make our definitions as restrictive as possible. We can alway relax them if it proves to be an issue.

Reply:
@gwrome 

I think we need to use SparseDtype here explicity  here

Reply:
@TomAugspurger I find the consrtuction of SparseDtype a bit odd here, rather shouldn't this has a subtype as we do for IntervalDtype?

Reply:
As far as this conversation goes I'd still propose what I mentioned earlier in terms of replacing `Type` with `str`, `float`, and `object` as I think that is a general purpose definition in the code base. I think the conversation around SparseDtype is orthogonal to this thread (OK to still address in this PR)

Reply:
@WillAyd ``Type`` is correct here or ``Type[int, float, object]``, not actul instances as indicated by @gwrome